### ISSUE TITLE
Posting GK at the same minute, with offline log, fail silently

### ISSUE THREAD

**Issue by kumy** [cgeo/cgeo#5182]
1. Prepare offline log of a series of caches
2. Post each saved log log along with GK visit

If in (1) two logs have been saved during the same hour-minute, the GK post will fail without notice to the user.

`[AsyncTask #4] GeokretyConnector.postLogTrackable: There is an entry with this date. Correct the date or the hour.`

**Comment by Lineflyer** [comment]
That is a "design bug" at GeoKrety.org isn't it?
They simply never accept dates on the same hour:minute.

**Comment by samueltardieu** [comment]
Yes, and this should be reported to GK. However, we should warn the user about the failed post, be it for this dubious reason or any other one.

**Comment by kumy** [comment]
> They simply never accept dates on the same hour:minute.

@Lineflyer exactly, but we should at least, notice that to the user. This is part of the refactor I talked yesterday. I think we're better doing that after release.

**Comment by kumy** [comment]
Do you really think it's a bug? Trackables are not omniscient ;)

**Comment by samueltardieu** [comment]
I don't think it's a question of being ubiquitous, but of being acted on by several people at roughly the same time. If a team discovers a cache and one of them takes the trackable, the other ones will likely mark it as visited, all at the same time.

**Comment by kumy** [comment]
Ok, I'll talk about that to Filips. :)
Or if someone with a "better english" than mine would like to explain that to him...
http://geokrety.org/kontakt.php -> Filips + cc g@g.c

**Comment by samueltardieu** [comment]
Can't you just him the URL of this issue?

**Comment by kumy** [comment]
done!

**Comment by filipsPL** [comment]
Hi All, I'm a Filip, the developer and maintainer of geokrety.org. Here I have a quick comment, where I agree with kummy: "Do you really think it's a bug? Trackables are not omniscient ;)" - indeed, we have deliberately done it this way. The only possible solution could be to increase "time resolution" of logs (to seconds), but I believe in 99.99% of cases it won't be useful. I will check if we can include easily this solution (or at least for API).

bests,
Filip

**Comment by samueltardieu** [comment]
Hi Filip. What problem does the "limit note to one per minute" solve? I would understand if you did that per user, to avoid double submissions and accidental reposts for example, but with multiple users, if makes no sense to me.

If GK become more popular, then you *will* have groups of geocachers visiting them. Just imagine one GK present at a geocaching.com mega-event, on a car for example. You will have thousands of people wanting to log it in a few hours. This is likely that some of them will attempt to do so a few seconds apart. Right now, only 300 people would be able to log it in a 5 hour long period, if they do so exactly every minute.

**Comment by filipsPL** [comment]
True. 
At www form, by default, the time is "current date" (for date) and 12:00 (for time). And to imply some chronography in the log, we had to apply some limitation at the log time, to prevent all logs from one date to be done at default time (12:00). I dont know if this is clear. Nevertheless, I belibe such limitation makes some sense.
The solution for the problem mentioned by samueltardieu would be the increase of logging resolution to seconds (microseconds). I will check the applicability of that.

**Comment by SammysHP** [comment]
Or don't use the time for the key at all …

**Comment by filipsPL** [comment]
... then how to sort logs and waypoints if we don't care about the time?

**Comment by kumy** [comment]
With geokrety/geokrety-website#1131 If the GK is marked as "non-collectible" then only a few log type are authorized: comment and met. In such case then the date time can be the same as another move.

Also the date time can be set at the "second" since GKv2. (it has been mentioned earlier in this thread)

The next steps looks to be on c:geo side to 
- properly display the API errors
- allow logging at the "seconds" level

🤷
